kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java

```diff
@@ -203,6 +206,48 @@ public void purgeOldInvalidSubscriptions(MotechEvent event) {
     }
 
     @Override
+    public void completePastDueSubscriptions() {
+
+        final DateTimeFormatter dateTimeFormatter = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss");
+        final DateTime currentTime = DateTime.now();
+        final DateTime oldestPregnancyStart = currentTime.minusDays(PREGNANCY_PACK_LENGTH_DAYS).withTimeAtStartOfDay();
+        final DateTime oldestChildStart = currentTime.minusDays(CHILD_PACK_LENGTH_DAYS).withTimeAtStartOfDay();
+
+        LOGGER.debug(String.format("Completing active pregnancy susbscriptions older than %s", oldestPregnancyStart));
+        LOGGER.debug(String.format("Completing active child susbscriptions older than %s", oldestChildStart));
+
+        @SuppressWarnings("unchecked")
+        SqlQueryExecution<Long> queryExecution = new SqlQueryExecution<Long>() {
+
+            @Override
+            public String getSqlQuery() {
+                String query = String.format(
+                        "UPDATE motech_data_services.nms_subscriptions AS s " +
+                        "JOIN motech_data_services.nms_subscription_packs AS sp " +
+                        "ON s.subscriptionPack_id_OID = sp.id " +
+                        "SET s.status = 'COMPLETED', s.endDate = '%s' " +
+                        "WHERE " +
+                        "s.status = 'ACTIVE' AND " +
```

Comment:
figured that pending activation state would never be possible since they get set to active on another query during target file generation? but, yes.. adding that might be safer, i think